components/wmtk_components/mesh_decimation/wmtk/components/mesh_decimation/internal/MeshDecimationOptions.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <nlohmann/json.hpp>
+
+namespace wmtk {
+namespace components {
+namespace internal {
+
+struct MeshDecimationOptions
+{
+    std::string input;
+    std::string output;
+    int64_t constrait_value;
+    double target_len;
+    std::string cell_constrait_tag_name;
+    std::vector<std::string> pass_through;
+};
+
+void to_json(nlohmann::json& j, MeshDecimationOptions& o);
```

Comment:
You could just use the macros here to handle the conversion but the functions are also fine.